From 1e1e76789f7931c27edadd763dac1e2917782059 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Fri, 8 May 2009 21:11:03 +0800
Subject: [PATCH 057/280] ENGR00112275 Fix SIM card build failure

Add rules to build .o and only build sim card library
for MX51

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 sim/Makefile | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/sim/Makefile b/sim/Makefile
index fbe4bd0..ee0e268 100644
--- a/sim/Makefile
+++ b/sim/Makefile
@@ -7,6 +7,10 @@ CFLAGS += $(INCLUDE)
 
 LIBNAME=libsim
 
+# list of platforms which want this test case
+INCLUDE_LIST:= IMX51
+
+ifeq ($(PLATFORM), $(findstring $(PLATFORM), $(INCLUDE_LIST)))
 all: $(LIBNAME).so
 
 install:
@@ -18,6 +22,13 @@ install:
 $(LIBNAME).so: $(OBJS)
 	$(CC) -shared -nostartfiles -Wl,-soname,$@ -o $@ $^ -lpthread
 
+%.o: %.c
+	$(CC) -D$(PLATFORM) $(INCLUDE) -Wall -O2 -c $^ -o $@
+
+else
+all install :
+endif
+
 .PHONY: clean
 clean:
 	@rm -f $(OBJS) $(LIBNAME).*
-- 
1.8.0

